azure/azure_service_fabric_cluster.go

```diff
@@ -0,0 +1,357 @@
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Cluster.Name"),
+			},
+			{
+				Name:        "tags",
+				Description: ColumnDescriptionTags,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Cluster.Tags"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Cluster.ID").Transform(idToAkas),
+			},
+
+			// Azure standard columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Cluster.Location").Transform(toLower),
+			},
+			{
+				Name:        "resource_group",
+				Description: ColumnDescriptionResourceGroup,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Cluster.ID").Transform(extractResourceGroupFromID),
+			},
+			{
+				Name:        "subscription_id",
+				Description: ColumnDescriptionSubscription,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Cluster.ID").Transform(idToSubscriptionID),
+			},
+		},
+	}
+}
+
+type ClusterInfo struct {
+	Cluster           servicefabric.Cluster
+	ClusterProperties ClusterPropertiesInfo
+}
+
+type ClusterPropertiesInfo struct {
+	AddOnFeatures                      *[]string
+	AvailableClusterVersions           interface{}
+	AzureActiveDirectory               interface{}
+	Certificate                        interface{}
+	CertificateCommonNames             interface{}
+	ClientCertificateCommonNames       interface{}
+	ClientCertificateThumbprints       interface{}
+	ClusterCodeVersion                 *string
+	ClusterEndpoint                    *string
+	ClusterID                          *string
+	ClusterState                       interface{}
+	DiagnosticsStorageAccountConfig    interface{}
+	EventStoreServiceEnabled           *bool
+	FabricSettings                     interface{}
+	ManagementEndpoint                 *string
+	NodeTypes                          interface{}
+	ProvisioningState                  interface{}
+	ReliabilityLevel                   interface{}
+	ReverseProxyCertificate            interface{}
+	ReverseProxyCertificateCommonNames interface{}
+	UpgradeDescription                 interface{}
+	UpgradeMode                        interface{}
+	VMImage                            *string
+}
+
+//// LIST FUNCTIONS
+
+func listServiceFabricClusters(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	session, err := GetNewSession(ctx, d, "MANAGEMENT")
+	if err != nil {
+		return nil, err
+	}
+	subscriptionID := session.SubscriptionID
+
+	clusterClient := servicefabric.NewClustersClient(subscriptionID)
+	clusterClient.Authorizer = session.Authorizer
+
+	result, err := clusterClient.List(ctx)
+	if err != nil {
```

Comment:
Can we add an error log here?
plugin.Logger(ctx).Error("listServiceFabricClusters", "list", err)